doc/compressed_transactions.md

```diff
@@ -0,0 +1,178 @@
+
+## 1. Abstract
+
+With this Transaction Compression Schema we use several methods to compress transactions,
+including dropping data and recovering it on decompression by grinding until we obtain
+valid signatures.
+
+The bulk of our size savings come from replacing the prevout of each input by a block
+height and index. This requires the decompression to have access to the blockchain, and
+also means that compression is ineffective for transactions that spend unconfirmed or
+insufficiently confirmed outputs.
+
+Even without compression, Taproot keyspends are very small: as witness data they
+include only a single 64/65-byte signature and do not repeat the public key or
+any other metadata. By using pubkey recovery, we obtain Taproot-like compression
+for legacy and Segwit transactions.
+
+The main applications for this schema are for steganography, satellite/radio broadcast, and
+other low bandwidth channels with a high CPU availability on decompression. We
+assume users have some ability to shape their transactions to improve their
+compressibility, and therefore give special treatment to certain transaction forms.
+
+## 2. Methods
+
+The four main methods to achieve a lower transactions size are:
+
+1. packing transaction metadata before the transaction and each of its inputs and
+outputs to determine the structure of the following data.
+2. replacing 32-bit numeric values with variable-length integers (VarInts).
+3. using compressed signatures and public key recovery upon decompression.
+4. replacing the 36-byte txid/vout pair with a blockheight and output index.
+
+Method 4 will cause the compressed transaction to be undecompressable if a block
+reorg occurs at or before the block it's included in. Therefore, we'll only compress
+the Txid if the transaction input is at least one hundred blocks old.
+
+## 3 Schema
+
+### 3.1 Primitives
+
+| Name             | Width     | Description |
+|------------------|-----------|-------------|
+| VarInt           | 1-5 Bytes | For 0-253, encode the value directly in one byte. For 254-65535, encode 254 followed by 2 little-endian bytes. For 65536-(2^32-1), encode 255 followed by 4 little-endian bytes. This is the same varint used to length-prefix strings in the ordinary transaction encoding. |
+| VarInt Flag      | 2 Bits    | 1, 2 or 3 indicate literal values. 0 indicates that the value will be encoded in a later VarInt. |
+| Bitstream VarInt | 1+ Bytes  | 7-bit little-endian encoding, with each 7-bit word encoded in a byte. The highest bit of each byte is 1 if more bytes follow, and 0 for the last byte. |
+| VLP-Bytestream   | 2+ Bytes  | A VarInt Length Prefixed Bytestream, Has a VarInt prefixed to determine the length. |
+
+### 3.2 General Schema
+
+| Name                           | Width           | Description |
+|--------------------------------|-----------------|-------------|
+| Transaction Metadata           | 7 Bits    | Information on the structure of the transaction. See Section 3.3. |
```

Comment:
In 6a117eac0af155610bb7bcfa30b6b878ca9628f4:

Is this actually 7 bits, or a full byte?

Reply:
I think we should add a "reserved" bit to get an even 8 bits. Trying to save a single bit will just make encoding/decoding harder.

Reply:
Its 7 bits of data bit its encoded into a single byte. Its actually a VarInt in case it needs to be bigger then 7 bits or even a full byte.